src/store/getters/state-getters.ts

```diff
@@ -445,7 +476,27 @@ export const getComponentsValidated = (state: StateIF): boolean => {
 
 /** Get resolution dates validity. */
 export const getIsResolutionDatesValid = (state: StateIF): boolean => {
-  if (hasShareStructureChanges(state) && getHasRightsOrRestrictions(state)) {
+  if (hasShareStructureChanged(state) && getHasRightsOrRestrictions(state)) {
     return getNewResolutionDates(state).length >= 1
-  } else return true
+  }
+  return true
+}
+
+/**
+ * Whether to show the fee summary.
+ * This is a safety check to ensure that fee summary component is not loaded
+ * until there is a valid filing type and entity code.
+ */
+export const showFeeSummary = (state: StateIF): boolean => {
+  const defaultFilingData = {
+    filingTypeCode: null as FilingCodes,
+    entityType: null as CorpTypeCd,
+    priority: false,
+    waiveFees: false
+  }
+  const haveFilingChange = (
+    (isCorrectionFiling(state) && hasCorrectionChanged(state)) ||
+    (isAlterationFiling(state) && hasAlterationChanged(state))
```

Comment:
I split up these two conditions for good measure. Theoretically if this is a correction filing then `hasAlterationChanged` should be False, and vice-versa, but I didn't want to assume that. (While refactoring this assumption failed. It may be OK now but safe is good.)